### PR TITLE
Change to ansible structure and add Vagrant

### LINKED ISSUE #17: Change structure to be more Ansible-Like
As discussed with @mattbaumann, it makes sense to use a more standardised Ansible setup.

### LINKED ISSUE #18: Add Vagrant
To test the Ansible deployment, we should add Vagrant (or something better or newer).

It is not possible to test the deployment in an easy way with docker, as docker-swarm-in-docker is not trivial...

### PR DESCRIPTION
**Pull request by fabianhauser**
Resolves #18 and resolves #17

### CODE REVIEW

File: roles/docker/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+  apt_repository:
+    repo: "deb [arch=amd64] https://download.docker.com/linux/debian {{ ansible_lsb.codename }} stable"
+    filename: "docker-stable"
+
+- name: Install docker
+  apt:
+    name: "docker-ce=18.03.1~ce-0~debian"
+    state: present
+    update_cache: yes
+  ignore_errors: yes
```
Comment (mattbaumann) [inline roles/docker/tasks/main.yml]: Is there a good reason to do so? If so, please comment.

File: roles/base/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  apt: pkg={{ item }} update-cache=yes cache_valid_time=3600
+  with_items:
+    - python
+    - python-apt
+    - python-pycurl
+    - python-psycopg2
+
+- name: Install system utilities
+  apt: pkg={{ item }} update-cache=yes cache_valid_time=3600
+  with_items:
```
Comment (mattbaumann) [inline roles/base/tasks/main.yml]: Recommend to use loop syntax, because longer supported syntax.

File: roles/docker/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    enabled: yes
+
+- name: Template demon config
+  copy:
+    src: "deamon.json"
+    dest: "/etc/docker/deamon.json"
+    owner: root
+    group: root
+    mode: u=rw, g=, o=
+  become: yes
```
Comment (mattbaumann) [inline roles/docker/tasks/main.yml]: Become used here, option with no effect

File: roles/docker/tasks/main.yml [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+  when: docker_daemon_config.changed
+
+# Init docker swarm on single server
+- name: Get docker info
+  shell: docker info
+  register: docker_info
+  changed_when: false
+
+- name: Init docker swarm mode
+  shell: docker swarm init
```
Comment (mattbaumann) [inline roles/docker/tasks/main.yml]: Shell used here, if this command is executed on 1 machine only, consider using command:

File: roles/docker/tasks/main.yml [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+  when: "docker_info.stdout.find('Swarm: inactive') != -1"
+
+- name: Get docker swarm manager token
+  shell: docker swarm join-token -q manager
+  register: manager_token
```
Comment (mattbaumann) [inline roles/docker/tasks/main.yml]: Has this command a side-effect or is it effect-less?

File: roles/docker/tasks/main.yml (hunk 1 above)
Comment (fabianhauser) [reply roles/docker/tasks/main.yml]: No, this was copy-pasted 😉 - removed

**Pull request merged**